facade/src/main/scala/reactST/reactTable/TableMaker.scala

```diff
@@ -22,91 +22,70 @@ case class TableMaker[D,
   TableInstanceD <: TableInstance[D], 
   ColumnOptsD <: ColumnOptions[D], 
   ColumnObjectD <: ColumnObject[D], 
-  State <: TableState[D] // format: on
+  TableStateD <: TableState[D] // format: on
 ](plugins: List[PluginHook[D]]) {
   import syntax._
 
-  /**
-   * Create an empty instance of TableOptsD.
-   */
-  def emptyOptions: TableOptsD = js.Dynamic.literal().asInstanceOf[TableOptsD]
+  private def emptyOptions: TableOptsD = js.Dynamic.literal().asInstanceOf[TableOptsD]
 
   /**
-   * Create a TableOptsD with a row id function and columns set.
+   * Create a TableOptsD. columns and data are required. Other options can be `set*`.
    */
-  // TODO When scalajs-react supports hooks, we can probably expose a List[Column[D]]
-  // in the API and useMemo.
-  def options(rowIdFn: D => String, columns: js.Array[Column[D]]): TableOptsD =
```

Comment:
I vaguely recall thinking that the `rowIdFn` was important enough that I put it here. But, I don't remember why. Is it just a performance thing? If the relatively large tables in the demo work fine, then I guess it isn't important.